### PR TITLE
Fix coverity issues

### PR DESCRIPTION
**Pull request by cmouse**
Fix some code quality issues.

### CODE REVIEW

**Comment by cmouse** [comment]
ready for review.

**Comment by cmouse** [comment]
The travis issue is gcc related

**Comment by cmouse** [comment]
I had to make rebase to fix some of the nits you had.

**Comment by Habbie** [comment]
LGTM, second review welcome (@pieterlexis?)

File: modules/opendbxbackend/odbxbackend.cc [hunk 1]
```diff
@@ -241,7 +241,7 @@ bool OdbxBackend::list( const DNSName& target, int zoneid, bool include_disabled
         {
         	DLOG( L.log( m_myname + " list()", Logger::Debug ) );
 
-        	m_qname.empty();
+                m_qname.clear();
```
Comment (pieterlexis) [inline modules/opendbxbackend/odbxbackend.cc]: one space too many?

File: modules/ldapbackend/ldapbackend.cc [hunk 2]
```diff
@@ -21,6 +21,9 @@ LdapBackend::LdapBackend( const string &suffix )
         	m_msgid = 0;
         	m_qname = "";
         	m_pldap = NULL;
+                m_ttl = 0;
```
Comment (pieterlexis) [inline modules/ldapbackend/ldapbackend.cc]: one space too many?

**Comment by pieterlexis** [comment]
apart from 2 whitespace nits, LGTM

File: modules/ldapbackend/ldapbackend.cc (hunk 2 above)
Comment (cmouse) [inline modules/ldapbackend/ldapbackend.cc]: actually a tab/space difference

File: pdns/dbdnsseckeeper.cc [hunk 3]
```diff
@@ -450,7 +450,7 @@ bool DNSSECKeeper::getPreRRSIGs(UeberBackend& db, const DNSName& signer, const D
                         // cerr<<"Got it"<<endl;
                         if (wildcardname.countLabels())
                                 rr.qname = qname;
-                        rr.d_place = (DNSResourceRecord::Place)signPlace;
+                        rr.d_place = static_cast<DNSResourceRecord::Place>(signPlace);
```
Comment (ahupowerdns) [inline pdns/dbdnsseckeeper.cc]: this should be DNSRecord::Place now, or signPlace should be of the right type anyhow. Is only a warning and not a real issue so we could fix this after the merge.

File: pdns/dnsname.cc [hunk 4]
```diff
@@ -24,6 +24,7 @@ std::ostream & operator<<(std::ostream &os, const DNSName& d)
 DNSName::DNSName(const char* p)
 {
   d_empty=false;
+  d_recurse=0;
```
Comment (ahupowerdns) [inline pdns/dnsname.cc]: we have a d_recurse in DNSName??? this is an abomination. We store hundreds of millions of DNSNames, none of which need d_recurse in storage.

File: pdns/dnsrecords.cc [hunk 5]
```diff
@@ -81,6 +81,12 @@ DNSResourceRecord::DNSResourceRecord(const DNSRecord &p) {
   qtype = p.d_type;
   ttl = p.d_ttl;
   setContent(p.d_content->getZoneRepresentation());
+  last_modified = 0;
+  signttl = 0;
+  domain_id = -1;
+  qclass = p.d_class;
+  d_place = static_cast<DNSResourceRecord::Place>(p.d_place);
```
Comment (ahupowerdns) [inline pdns/dnsrecords.cc]: s/DNSResourceRecord/DNSRecord/

File: pdns/dnssecsigner.cc [hunk 6]
```diff
@@ -117,7 +117,7 @@ void addSignature(DNSSECKeeper& dk, UeberBackend& db, const DNSName& signer, con
     else
       rr.ttl=signTTL;
     rr.auth=false;
-    rr.d_place = (DNSResourceRecord::Place) signPlace;
+    rr.d_place = static_cast<DNSResourceRecord::Place> (signPlace);
```
Comment (ahupowerdns) [inline pdns/dnssecsigner.cc]: s/DNSResourceRecord/DNSRecord/ or again fix signPlace to be the right type

File: pdns/dnssecsigner.cc [hunk 7]
```diff
@@ -230,7 +230,7 @@ void addRRSigs(DNSSECKeeper& dk, UeberBackend& db, const set<DNSName>& authSet,
     else
       signTTL = pos->ttl;
     origTTL = pos->ttl;
-    signPlace = (DNSPacketWriter::Place) pos->d_place;
+    signPlace = static_cast<DNSPacketWriter::Place>(pos->d_place);
```
Comment (ahupowerdns) [inline pdns/dnssecsigner.cc]: or maybe we should revamp our **THREE** place enums into one one day ;-)

File: pdns/dnsrecords.cc (hunk 5 above)
Comment (cmouse) [inline pdns/dnsrecords.cc]: target is DNSResourceRecord::Place

File: pdns/secpoll-auth.cc [hunk 8]
```diff
@@ -171,9 +171,6 @@ void doSecPoll(bool first)
       L<<Logger::Warning<<"Could not retrieve security status update for '" + pkgv + "' on '"+query+"', RCODE = "<< RCode::to_s(res)<<endl;
     else
       L<<Logger::Warning<<"Not validating response for security status update, this a non-release version."<<endl;
-
-    if(security_status == 1) // it was ok, not it is unknown
```
Comment (ahupowerdns) [inline pdns/secpoll-auth.cc]: what does this change?

Comment (cmouse) [inline pdns/secpoll-auth.cc]: dead code. it's never reached. we agreed this with @habbie before.

**Pull request merged**